crates/ruff_cli/src/diagnostics.rs

```diff
@@ -243,7 +243,8 @@ pub(crate) fn lint_path(
                             write(path, transformed.as_bytes())?;
                         }
                         SourceKind::IpyNotebook(notebook) => {
-                            notebook.write(path)?;
+                            let mut writer = BufWriter::new(File::create(path)?);
+                            notebook.write(&mut writer)?;
                         }
```

Comment:
Can you explain what this change is about?

Reply:
I believe it's to merge the `write` and `write_inner` function on `Notebook`. `write` would require a `Path` while `write_inner` would use a `impl Write`. Now, there's only a single public function `write` which takes in a `impl Write`.

The main reason for the separation was in tests where the `write_inner` was being utilized to avoid changing the file on disk and to just use a `Vec` as a writer.

Hope this helps otherwise Charlie can expand as well :)

Reply:
Yeah, I think it's more flexible to expose an interface that takes `impl Write`.

Reply:
We may want to consider a `&dyn Write` to avoid monomorphizing this for every write.